src/gt4py/next/iterator/embedded.py

```diff
@@ -1443,14 +1417,9 @@ def field_setitem(self, named_indices: NamedFieldIndices, value: Any):
 
 
 def as_tuple_field(field: tuple | TupleField) -> TupleField:
-    assert can_be_tuple_field(field)
-
-    if is_tuple_of_field(field):
-        assert not isinstance(field, TupleField)
-        return TupleOfFields(tuple(_wrap_field(f) for f in field))
-
-    assert isinstance(field, TupleField)  # e.g. field of tuple is already TupleField
-    return field
+    assert is_tuple_of_field(field)
+    assert not isinstance(field, TupleField)
```

Comment:
This removes support for structured dtypes (tests where already removed earlier in this PR).